pkg/controllerutils/update.go

```diff
@@ -79,3 +82,70 @@ func TypedCreateOrUpdate(ctx context.Context, c client.Client, scheme *runtime.S
 
 	return controllerutil.OperationResultUpdated, c.Update(ctx, obj)
 }
+
+// TryUpdateStatus tries to apply the given transformation function onto the given object, and to update its
```

Comment:
WDYT about moving this to the worker actuator and making this unexported to prevent an increase of usages again?